specification/data-faas.md

```diff
@@ -0,0 +1,29 @@
+# Semantic conventions for FaaS spans
+
+This document defines how to describe an instance of a function that runs without provisioning or managing of servers (also known as serverless) with spans.
+
+Span `name` should be set to the function name being executed. Depending on the trigger of the function, additional attributes MUST be set. For example, an HTTP trigger MUST follow the [HTTP Server semantic conventions](data-http.md#http-server-semantic-conventions).
```

Comment:
I'm not sure if it is always possible to follow e.g. HTTP Server semantic conventions. For example an AWS lambda function may be triggered using AWS SDK which in turn uses HTTP.
Or AWS Lambda function triggered by AWS SQS. Function is triggered once but may get a bulk of messages. One would expect to see one span per message (linked to the corresponding sender) and not one per lambda function call (not linkable to anyone).

Reply:
It is indeed true that the SDK internally uses an HTTP call to trigger a lambda. However, the lambda itself might not be configured to process an HTTP request. This attribute is not for how a lambda is called but more on the type of event the lambda is configured to answer to.

Reply:
Ah ok. maybe write `Depending on the function input data` or similar?

Reply:
I understand the confusion. I agree, the wording could be improved